perception/vis/vis.py

```diff
@@ -1,58 +1,68 @@
 import argparse
-from FrameWrapper import FrameWrapper
+from perception.vis.FrameWrapper import FrameWrapper
 import cv2 as cv
-from window_builder import Visualizer
-import cProfile as cp
-import pstats
-
-# Parse arguments
-parser = argparse.ArgumentParser(description='Visualizes perception algorithms.')
-parser.add_argument(
-    '--data', default='webcam', type=str
-)
-parser.add_argument('--algorithm', type=str)
-parser.add_argument('--save_video', action='store_true')
-args = parser.parse_args()
-
-# Get algorithm module
-exec("from TestTasks.{} import {} as Algorithm".format(args.algorithm, args.algorithm))
-
-# Initialize image source
-data_sources = [args.data]
-data = FrameWrapper(data_sources, 0.25)
-
-algorithm = Algorithm()
-window_builder = Visualizer(algorithm.var_info())
-video_frames = []
-# Main Loop
-def main():
+from perception.vis.Visualizer import Visualizer
+import cProfile
+
+
+def run(data_sources, algorithm, save_video=False):
+    out = None
+    window_builder = Visualizer(algorithm.variables)
+    data = FrameWrapper(data_sources, 0.25)
+    frame_count = 0
+    paused = False
+    speed = 1
+
     for frame in data:
+        if frame_count % speed == 0 and not paused:
```

Comment:
If the program is paused, the loop will still run and a lot of frames will be skipped